files/msgdata/msg.mk

```diff
@@ -7,11 +7,21 @@ MSGFILE_GMM := $(sort $(wildcard $(MSGDATA_MSG_DIR)/*.gmm) $(TRNAME_GMM))
 MSGFILE_BIN := $(patsubst %.gmm,%.bin,$(MSGFILE_GMM))
 MSGFILE_H := $(patsubst %.gmm,%.h,$(MSGFILE_GMM))
 
+TRAINER_JSON := files/poketool/trainer/trainers.json
+TRNAME_TEMPLATE := files/poketool/trainer/trname.json.txt
+
 FIRST_MSG_H_GEN := $(MSGDATA_DIR)/headers.done
 TOUCH_ONCE      := $(MSGDATA_DIR)/touch_once.sh
 
+$(TRNAME_GMM):
+	$(JSONPROC) $(TRAINER_JSON) $(TRNAME_TEMPLATE) $(TRNAME_GMM)
+	$(SED) -i 's/&/&amp;/g' $(TRNAME_GMM)
+
 $(MSGDATA_MSG_DIR).narc: %.narc: $(MSGFILE_BIN)
 
+# extremely ugly hack to get it actually building
+$(MSGDATA_MSG_DIR)/msg_0729.bin: $(TRNAME_GMM)
```

Comment:
what's this for? if we can solve this issue it will likely prevent further issues in the future

Reply:
since the bin require the gmm to be built, it needs to be marked as a dependency of it, I attempted to make it a dependency of all the bins so it's built before a single one, but either I dont understand make well enough or some fuckery was going on

Reply:
ended up deciding this will be a project for the future, as I'm not sure the exactly circumstances of more generated gmms would be, so will be addressed when/if we end up building more of then

Reply:
That's ok lol but I'm not merging this until the issue is resolved Aka you don't do a gamefreak and leave in hard coded work

Reply:
*screams*